pymc/model_graph.py

```diff
@@ -49,6 +49,9 @@ class PlateMeta:
     def __hash__(self):
         return hash((self.names, self.sizes))
 
+    def __bool__(self) -> bool:
+        return len(self.sizes) > 0 or len(self.names) > 0
```

Comment:
What is a plate without names?

Reply:
```python
with pm.Model(): 
    pm.Normal("y", shape=3)
```
y has sizes but no dim names. 
Currently creates Plate(meta=PlateMeta(names=(), sizes=(5, )), variables=[NodeMeta(var=y, node_type=...)])

Think there should be some cases to test now that this logic is exposed. Will be much easier to confirm

Reply:
names here are dim names

Reply:
What happens for a Deterministic with `dims=("test_dim", None)`? Apparently we still allow None dims for things that are not RVs

Reply:
That `y` should be `names=(None,)` ?

Reply:
I'm thinking of `pm.Deterministic("x", np.zeros((3, 3)), dims=("hello", None))` and `pm.Deterministic("y", np.zeros((3, 3)), dims=(None, "hello")`. We don't want to put those in the same plate because dims can't be repeated, so they are definitely different things?

Reply:
Added a test. I had to wrap the data in as_tensor_variable or I'd get an error saying the data needs name attribute